php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1342 @@
+      list = SeparatedList.empty();
+    }
+
+    return new FunctionCallTreeImpl(openParenthesis, list, closeParenthesis);
+  }
+
+  public MemberAccessTree classMemberAccess(InternalSyntaxToken token, Tree member) {
+    return new MemberAccessTreeImpl(Kind.CLASS_MEMBER_ACCESS, token, member);
+  }
+
+  public ExpressionTree objectDimensionalList(ExpressionTree variableName, Optional<List<ArrayAccessTree>> dimensionalOffsets) {
+    ExpressionTree result = variableName;
+
+    for (ArrayAccessTree arrayAccess : optionalList(dimensionalOffsets)) {
+      result = ((ArrayAccessTreeImpl) arrayAccess).complete(result);
+    }
+
+    return result;
+  }
+
+  public IdentifierTree variableName(InternalSyntaxToken token) {
+    return new IdentifierTreeImpl(token);
+  }
+
+  public MemberAccessTree objectMemberAccess(InternalSyntaxToken accessToken, ExpressionTree member) {
+    return new MemberAccessTreeImpl(Kind.OBJECT_MEMBER_ACCESS, accessToken, member);
+  }
+
+  public ExpressionTree memberExpression(ExpressionTree object, Optional<List<ExpressionTree>> memberAccesses) {
+    ExpressionTree result = object;
+
+    for (ExpressionTree memberAccess : optionalList(memberAccesses)) {
+      if (memberAccess.is(Kind.OBJECT_MEMBER_ACCESS, Kind.CLASS_MEMBER_ACCESS)) {
+        result = ((MemberAccessTreeImpl) memberAccess).complete(result);
+
+      } else if (memberAccess.is(Kind.ARRAY_ACCESS)) {
+        result = ((ArrayAccessTreeImpl) memberAccess).complete(result);
+
+      } else if (memberAccess.is(Kind.FUNCTION_CALL)) {
+        result = ((FunctionCallTreeImpl) memberAccess).complete(result);
+      }
+    }
+
+    return result;
+  }
+
+  public VariableTree lexicalVariable(Optional<InternalSyntaxToken> ampersandToken, VariableIdentifierTree variableIdentifier) {
+    return ampersandToken.isPresent()
+      ? new ReferenceVariableTreeImpl(ampersandToken.get(), variableIdentifier)
+      : variableIdentifier;
+  }
+
+  public LexicalVariablesTree lexicalVariables(
+      InternalSyntaxToken useToken, InternalSyntaxToken openParenthesis,
+      VariableTree variable, Optional<List<Tuple<InternalSyntaxToken, VariableTree>>> variableRest,
+      InternalSyntaxToken closeParenthesis
+  ) {
+    return new LexicalVariablesTreeImpl(useToken, openParenthesis, separatedList(variable, variableRest), closeParenthesis);
+  }
+
+  public FunctionCallTree internalFunction1(InternalSyntaxToken issetToken, InternalSyntaxToken openParenthesis, ExpressionTree expression, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>> expressionRest, InternalSyntaxToken closeParenthesis) {
+    return new FunctionCallTreeImpl(
+      new IdentifierTreeImpl(issetToken),
+      openParenthesis,
+      separatedList(expression, expressionRest),
+      closeParenthesis);
+  }
+
+  public FunctionCallTree internalFunction2(InternalSyntaxToken functionNameToken, InternalSyntaxToken openParenthesis, ExpressionTree expression, InternalSyntaxToken closeParenthesis) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 186 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)